danger-klaxit/lib/klaxit/config_sorter.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module Klaxit
+  class ConfigSorter
+    attr_reader :file
+
+    # Regexp that detects a group, and returns its name as first match.
+    PARENT_RE = /\A(\w+):/.freeze
+    # Regexp that detects any child key in a group, and returns its name as
+    # first match.
+    CHILD_RE = /\A  (\w+):/.freeze
+
+    def initialize(file)
+      @file = file
+    end
+
+    # @return [String] the sorted file
+    def sorted_file
+      @sorted_file ||= parse
+    end
+
+    # Check if there is a change in the +file+, in other words: check if the
+    # order was not correct.
+    def changed?
+      @changed ||= sorted_file != IO.read(file)
+    end
+
+    # Get a git diff between the current file and sorted file.
+    # @return [String]
+    def diff
+      require "tempfile"
+      new_file = Tempfile.new(["new_config.", ".yml"])
+      new_file.write(sorted_file)
+      result = `git diff -- #{file} #{new_file.path}`.gsub(
+        no_slash_beginning(new_file.path),
+        no_slash_beginning(file)
+      )
+    ensure
+      new_file.close
+      new_file.unlink
+      result
+    end
+
+    private
+
+    def parse
+      current_group = nil
+      data_for_group = Hash.new { |h, k| h[k] = String.new }
```

Comment:
What is this init form exactly ?

Reply:
The kind of init form that is not buggy! https://medium.com/klaxit-techblog/a-headache-in-ruby-hash-default-values-bf2706660392